controllers/urls.js

```diff
@@ -1,16 +1,51 @@
-const express = require('express')
-const Url = require('../models/Url')
+const express = require('express');
+const { validationResult } = require('express-validator');
+const statusCodes = require('http-status-code');
+const Url = require('../models/Url');
+const generateRandomString = require('../middleware/randomString');
 
-const createShortUrl = async(req, res) => {
-    const url = await Url.create({...req.body})
-    res.json(url)
-}
+const createShortUrl = async (req, res) => {
+  const { customName, originalUrl } = req.body;
+  let shortUrls;
 
-const customShortUrl = async(req, res)  => {
-    res.send('custom name for short url')
-}
+  try {
+    if (customName) {
+      const customNameWithDashes = customName.split(' ').join('-');
+      shortUrls = `https://shortit/${customNameWithDashes}`;
+      const existingCustomName = await Url.findOne({ customName: customNameWithDashes });
+      if (existingCustomName) {
+        return res.status(400).json({ errors: 'Custom name already exists' });
```

Comment:
Another improvement we can make is to extract the HTTP error code to a constant file and reuse it.

```
const HTTP_ERRORS = {
  OK: 200,
  BAD_REQUEST: 400,
  INTERNAL_SERVER_ERROR: 500
}
```

```suggestion
        return res.status(HTTP_ERROR.BAD_REQUEST).json({ errors: 'Custom name already exists' });
```

Reply:
done on 01578a4